allure-grpc/src/main/java/io/qameta/allure/grpc/AllureGrpc.java

```diff
@@ -0,0 +1,162 @@
+                ));
+                try {
+                    GrpcRequestAttachment rpcRequestAttach = GrpcRequestAttachment.Builder
+                            .create("gRPC request", method.getFullMethodName())
+                            .setBody(jsonPrinter.print((MessageOrBuilder) message))
+                            .build();
+                    processor.addAttachment(rpcRequestAttach, new FreemarkerAttachmentRenderer(requestTemplatePath));
+                    super.sendMessage(message);
+                } catch (InvalidProtocolBufferException e) {
+                    log.warn("Can`t parse gRPC request", e);
+                } catch (Throwable e) {
+                    Allure.getLifecycle().updateStep((s) ->
+                            s.setStatus(ResultsUtils.getStatus(e).orElse(Status.BROKEN))
+                                    .setStatusDetails(ResultsUtils.getStatusDetails(e).orElse(null))
+                    );
+                    Allure.getLifecycle().stopStep(stepUuid);
+                    stepUuid = null;
+                }
+            }
+
+            @Override
+            public void start(Listener<RespT> responseListener, Metadata headers) {
+                ClientCall.Listener<RespT> listener = new ForwardingClientCallListener<>() {
+                    @Override
+                    protected Listener<RespT> delegate() {
+                        return responseListener;
+                    }
+
+                    @Override
+                    public void onClose(io.grpc.Status status, Metadata trailers) {
+                        if (parsedResponses.size() == 1) {
+                            GrpcResponseAttachment rpcResponseAttach = GrpcResponseAttachment.Builder
+                                    .create("gRPC response")
+                                    .setBody(parsedResponses.iterator().next())
+                                    .build();
+                            processor.addAttachment(rpcResponseAttach, new FreemarkerAttachmentRenderer(responseTemplatePath));
+                        } else if (parsedResponses.size() > 1) {
+                            GrpcResponseAttachment rpcResponseAttach = GrpcResponseAttachment.Builder
+                                    .create("gRPC response (collection of elements from Server stream)")
+                                    .setBody("[" + String.join(",\n", parsedResponses) + "]")
+                                    .build();
+                            processor.addAttachment(rpcResponseAttach, new FreemarkerAttachmentRenderer(responseTemplatePath));
+                        }
+
+                        if (status.isOk()) {
+                            Allure.getLifecycle().updateStep(stepUuid, (step) -> step.setStatus(Status.PASSED));
+                        } else {
+                            Allure.getLifecycle().updateStep(stepUuid, (step) -> step.setStatus(Status.FAILED));
```

Comment:
I'm not entirely sure if the step should be considered as a failed step. What if we are testing some logic that should handle messages with some questionable status, for example custom implementation of "backpreasure"?

Reply:
Added optional flag `markStepFailedOnNonZeroCode`
by default it will be `true` for my opinion.